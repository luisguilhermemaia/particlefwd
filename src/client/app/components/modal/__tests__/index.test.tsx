import '@testing-library/jest-dom/extend-expect';
import { RenderResult, render } from '@testing-library/react';
import React from 'react';
import { GalleryModal } from '..';

describe('GalleryModal', () => {
  let renderResult: RenderResult;
  const images = [
    {
      uri: 'https://particleforward.com/api/challenge/assets/image1',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image1',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image2',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image3',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image4',
      set: 'fe4cfedffdffffffff',
    },
  ];
  const doRender = () => {
    renderResult = render(
      <GalleryModal
        closeModal={jest.fn()}
        findPrev={jest.fn()}
        findNext={jest.fn()}
        hasPrev={false}
        hasNext={true}
        src={images[0].uri}
      />,
    );
  };

  it('renders correctly with modal component', () => {
    doRender();

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });
});
