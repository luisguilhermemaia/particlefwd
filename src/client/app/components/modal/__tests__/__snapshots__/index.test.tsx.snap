// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryModal renders correctly with modal component 1`] = `
<div>
  <div>
    <div
      class="modal__StyledOverlay-sc-512xae-1 dfXBME"
    />
    <div
      class="modal__StyledContainer-sc-512xae-0 kyDfPY"
    >
      <div
        class="modal__StyledBody-sc-512xae-5 hkFgGp"
      >
        <a
          class="modal__StyledAnchorClose-sc-512xae-2 hQUDcx"
          href="#"
        >
          ×
        </a>
        <a
          class="modal__StyledAnchorNext-sc-512xae-3 hjJdIc"
          href="#"
        >
          ›
        </a>
        <img
          src="https://particleforward.com/api/challenge/assets/image1_27.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
